sputnikdao-factory2/src/lib.rs

```diff
@@ -109,30 +80,115 @@ impl SputnikDAOFactory {
             false
         }
     }
+
+    /// Tries to upgrade given account created by this factory to the latest code.
+    pub fn upgrade(&self, account_id: AccountId) {
```

Comment:
If not set otherwise, the factory itself will pay for all the upgrades right? And if we want to change that, we change the owner. Do I understand correctly?

Reply:
Generally, it is expected that storage is already covered before this is called (because this function is not payable) - given we are replacing contract in-place.
Whoever called the factory for the upgrade will be paying gas fees for this execution indeed.